### PR TITLE
users: Add preview of avatar when selecting a file

### LINKED ISSUE #101: Profile picture upload does not update live
When uploading a new profile picture from the profile page (i.e. `https://prologin.org/user/###/edit`)

the view does not update the new image in the profile picture section. Although, saving changes does aplly the change.

Not seeing the changes before the save is a bit confusing .

### PR DESCRIPTION
**Pull request by cafehaine**
Fixes #101

### CODE REVIEW

File: prologin/users/static/js/user-edit.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+document.addEventListener("DOMContentLoaded", init);
+
+function init()
+{
+    let formGroupIds = ["id_picture", "id_avatar"];
+
+    for (let i = 0; i < formGroupIds.length; i++)
```
Comment (zopieux) [inline prologin/users/static/js/user-edit.js]: If you use ecmascript 6 `let` you can also use ecmascript 6 array iterators.

File: prologin/users/static/js/user-edit.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        let form = document.getElementById("div_" + formGroupIds[i]);
+        if (form != null)
```
Comment (zopieux) [inline prologin/users/static/js/user-edit.js]: What about `continue` on null so you can remove one depth level.

**Review by zopieux — DISMISSED** [review]
Thx.

File: prologin/users/static/js/user-edit.js (hunk 2 above)
Comment (cafehaine) [reply prologin/users/static/js/user-edit.js]: cannot use it since I'm now using forEach

File: prologin/users/static/js/user-edit.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+function init() {
+  let formGroupIds = ["id_picture", "id_avatar"];
+
+  formGroupIds.forEach(id => {
+    let form = $("#div_" + id);
+
+    if (form != null) {
+      $("#" + id).change(function (e) {
+        let files = this.files;
```
Comment (seirl) [inline prologin/users/static/js/user-edit.js]: nit: useless to redefine a variable here, just use this.files.length and this.files.[0] in the rest of the file

File: prologin/users/static/js/user-edit.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+document.addEventListener("DOMContentLoaded", init);
+
+function init() {
+  let formGroupIds = ["id_picture", "id_avatar"];
+
+  formGroupIds.forEach(id => {
```
Comment (seirl) [inline prologin/users/static/js/user-edit.js]: function(id) {

**Review by seirl — CHANGES_REQUESTED** [review]

File: prologin/users/static/js/user-edit.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+document.addEventListener("DOMContentLoaded", init);
```
Comment (seirl) [inline prologin/users/static/js/user-edit.js]: $(function() {
   ... code of init() here ...
}

**Review by seirl — CHANGES_REQUESTED** [review]

File: prologin/users/static/js/user-edit.js (hunk 5 above)
Comment (cafehaine) [reply prologin/users/static/js/user-edit.js]: cannot do it, the js is loaded before jquery since it is included in the form media by django

**Review by seirl — APPROVED** [review]

**Comment by seirl** [comment]
Thanks, merged.

**Pull request merged**